src/App.tsx

```diff
@@ -34,29 +35,43 @@ export const App: React.FC = () => {
       switch (loadType) {
         case LoadType.FIRST5:
           goodsFromServer = await get5First();
+
+          setDataLoadedFirst5(true);
+          setDataLoadedOnlyRed(false);
+          setDataLoadedAll(false);
+
           break;
 
         case LoadType.ONLYRED:
           goodsFromServer = await getRedGoods();
+
+          setDataLoadedOnlyRed(true);
+          setDataLoadedFirst5(false);
+          setDataLoadedAll(false);
+
           break;
 
         default:
           goodsFromServer = await getAll();
+
+          setDataLoadedAll(true);
+          setDataLoadedOnlyRed(false);
+          setDataLoadedFirst5(false);
+
+          break;
       }
```

Comment:
```suggestion
      goodsFromServer = await loadCallback();
```

Can be simple as this